homeassistant/components/switch/x10.py

```diff
@@ -0,0 +1,88 @@
+"""
+Support for X10 switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.x10/
+"""
+import logging
+from subprocess import check_output, CalledProcessError, STDOUT
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_NAME, CONF_ID, CONF_DEVICES)
+from homeassistant.components.switch import (
+    SwitchDevice, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [
+        {
+            vol.Required(CONF_ID): cv.string,
+            vol.Required(CONF_NAME): cv.string,
+        }
+    ]),
+})
+
+
+def x10_command(command):
+    """Execute X10 command and check output."""
+    return check_output(['heyu'] + command.split(' '), stderr=STDOUT)
+
+
+def get_status():
+    """Get on/off status for all x10 units in default housecode."""
+    output = check_output('heyu info | grep monitored', shell=True)
+    return output.decode('utf-8').split(' ')[-1].strip('\n()')
+
+
+def get_unit_status(code):
```

Comment:
This command will be called once per device per update round. This means that if you have 10 devices, we call this command 10 times within a second, although all information is there in the first call.

You need to setup a shared data resource that will help you throttle the calls. Have a look at [Bitcoin data](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/bitcoin.py#L170) for an exmaple of how to accomplish this.